pkg/build/poller.go

```diff
@@ -0,0 +1,142 @@
+// Copyright 2024 Daytona Platforms Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+package build
+
+import (
+	"fmt"
+
+	"github.com/daytonaio/daytona/pkg/gitprovider"
+	"github.com/daytonaio/daytona/pkg/server/gitproviders"
+	log "github.com/sirupsen/logrus"
+)
+
+type PollerConfig struct {
+	Scheduler          IScheduler
+	Interval           string
+	BuilderFactory     IBuilderFactory
+	BuildStore         Store
+	GitProviderService gitproviders.IGitProviderService
+}
+
+type IPoller interface {
```

Comment:
I believe this should be its own abstraction in its own package.

`type Poller` should then be `type BuildPoller`.

`IPoller` should be a generic type and `type Poller` should myb be an abstract poller that has start and stop, while `BuildPoller` would only implement the `Poll` method then.

This would allow us to reuse the Poll interface and abstraction for other cases in the future.